presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java

```diff
@@ -263,6 +300,77 @@ private static MethodDefinition generateProjectColumnarMethod(
     {
         Parameter session = arg("session", ConnectorSession.class);
         Parameter page = arg("page", Page.class);
+        Parameter pageBuilder = arg("pageBuilder", PageBuilder.class);
```

Comment:
I can try to extract common parts somehow, but I am not sure if it will be more elegant.